mbrtowc: set output to NULL in case of empty input string like glibc does.

 https://bugs.busybox.net/show_bug.cgi?id=1471

--- a/libc/misc/wchar/wchar.c	2008-12-02 20:15:35.000000000 +0300
+++ b/libc/misc/wchar/wchar.c	2010-11-22 20:23:52.000000000 +0300
@@ -295,6 +295,8 @@ size_t mbrtowc(wchar_t *__restrict pwc,
 		n = 1;
 	} else if (*s == '\0') {
     /* According to the ISO C 89 standard this is the expected behaviour.  */
+		if (pwc)
+			*pwc = 0;
 		return 0;
 	} else if (!n) {
 		/* TODO: change error code? */
